From 7fdf65833eb239955cf2715dac96ea804770896c Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Mon, 1 Jun 2026 00:20:07 +0500
Subject: [PATCH 15317/15333] dt-bindings: display: panel: Add Samsung EA8074

Add a binding for the Samsung EA8074 display driver IC.
The panel is found on the Xiaomi Mi 8 (dipper) and Comma AI box.

Based on work of: Max Furman <taygoth@gmail.com>

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 .../display/panel/samsung,ea8074.yaml         | 78 +++++++++++++++++++
 1 file changed, 78 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/samsung,ea8074.yaml

diff --git a/Documentation/devicetree/bindings/display/panel/samsung,ea8074.yaml b/Documentation/devicetree/bindings/display/panel/samsung,ea8074.yaml
new file mode 100644
index 000000000000..c2e8c246bd95
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/samsung,ea8074.yaml
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/samsung,ea8074.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Samsung EA8074 AMOLED DDIC
+
+maintainers:
+  - FIXME
+
+description: |
+  The Samsung EA8074 is display driver IC with connect panel.
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    items:
+      - enum:
+           # Samsung 6.01 inch, 1080x2160 pixels
+          - samsung,ea8074-ams601nt01
+           # Samsung 6.21 inch, 1080x2248 pixels
+          - samsung,ea8074-ams621rc
+           # Samsung 5.9 inch, 1080x2160 pixels
+          - samsung,ea8074-l590nt06
+      - const: samsung,ea8074
+
+  reg:
+    maxItems: 1
+
+  elvdd-supply:
+    description: EL driving positive voltage supply
+
+  elvss-supply:
+    description: EL driving negative voltage supply
+
+  vddio-supply:
+    description: I/O voltage supply
+
+required:
+  - compatible
+  - reg
+  - reset-gpios
+  - elvdd-supply
+  - elvss-supply
+  - vddio-supply
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    dsi {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        panel@0 {
+            compatible = "samsung,ea8074-ams621rc", "samsung,ea8074";
+            reg = <0>;
+
+            reset-gpios = <&tlmm 6 GPIO_ACTIVE_LOW>;
+
+            vddio-supply = <&vreg_l14a_1p8>;
+            elvdd-supply = <&lab>;
+            elvss-supply = <&ibb>;
+
+            port {
+                panel_in: endpoint {
+                    remote-endpoint = <&dsi0_out>;
+                };
+            };
+        };
+    };
+
+...
-- 
2.55.0

